todo-list/src/components/todos/todoList.tsx

```diff
@@ -0,0 +1,72 @@
+import { useAddTodo, useTodos } from "@/hooks/todos/useTodo";
+import { useState } from "react";
+import { Input } from "../ui/input";
+import { Button } from "../ui/button";
+import { TodoItem } from "./todoItem";
+import { Todo } from "@/types/todos/todos.type";
+import { useTodoStore } from "@/hooks/store/useStore";
+
+export function TodoList() {
+  const zustandTodos = useTodoStore((state) => state.todos);
+  const { todos: queryTodos, isLoading, isError } = useTodos();
+  const { handleAddTodo, isLoading: isAdding } = useAddTodo();
+  const [newTodo, setNewTodo] = useState("");
+
+  const combinedTodos = [
+    ...zustandTodos, 
+    ...(queryTodos?.todos || []) 
+  ];

```

Comment:
This should use `useMemo` to avoid changing `combinedTodos` in each render, which in turn cause multiple re-renders